src/TestSuite/Fixture/FixtureManager.php

```diff
@@ -256,7 +256,9 @@ protected function _setupTable(
 
         $hasSchema = $fixture instanceof TableSchemaAwareInterface && $fixture->getTableSchema() instanceof TableSchema;
 
-        if (($drop && $exists) || ($exists && $hasSchema)) {
+        if ($exists && !$fixture->import && !$fixture->fields) {
```

Comment:
The interface for fixtures doesn't have these properties. You should be able to use the $hasSchema variable from like 257 instead as it represents whether or not the fixture defined any kind of schema.

Reply:
Oh ok, I see. But using $hasSchema isn't working because the fixture's _schema property is always being set, even when both the imports and fields properties are empty, it's set by this method: `Cake\TestSuite\Fixture\TestFixture::_schemaFromReflection()`

Reply:
Maybe we can just change `TestFixture::drop()` and truncate if there are no fields set instead of only returning true. That seems to be the expected behavior.

This does tie the FixtureManager and TestFixture implementation together because we expect `drop()` before `create()` if the table exists. I don't think we want to truncate in `create()` as well.

Reply:
> Is there any way to determine from the TableSchema object if the schema was built directly from a datasource?

Right now there isn't we would need to expand the TableSchemaAwareInterface to have that if if we need it.